uxarray/grid/validation.py

```diff
@@ -65,3 +65,25 @@ def _check_area(self):
     else:
         print("-No face area is close to zero.")
         return True
+
+
+def _check_normalization(grid):
+    """Checks whether all the cartesiain coordinates are normalized."""
+
+    if "node_x" in grid._ds:
+        if not (
+            np.isclose((grid.node_x**2 + grid.node_y**2 + grid.node_z**2), 1.0)
```

Comment:
@hongyuchen1030 

I've updated the check here to account for small floating point differences when checking if the sum of squares adds to one. Does this look good as a check, or should I include anything else?

Reply:
Thanks for your work!
1. I feel like we should set the error tolerance for the isclose function
2. As I mentioned before, this kind of check will loop over the entire grid and do the following: 

Step1: Calculate grid.node_x**2 + grid.node_y**2 + grid.node_z**2 and see if their results are one

Step2: if the results are not one, Calculate grid.node_x**2 + grid.node_y**2 + grid.node_z**2 again so it can know the norm for the following normalization step.

Can we just store the norm from Step1 until we either: 1. the step 1 return true 2. Or we finish the step 2 normalization?

Reply:
> Thanks for your work!
> 
> 1. I feel like we should set the error tolerance for the isclose function
np.close has:
rtol (relative tolerance): Default is 1e-05
atol (absolute tolerance): Default is 1e-08

what do you wnat to set it to? we can set it as:

`result = np.isclose(result, 1, rtol=1e-06, atol=1e-09)`

Reply:
> As I mentioned before, this kind of check will loop over the entire grid and do the following:

Great suggestion and thanks for the reminder, forgot to include this.

Reply:
> I feel like we should set the error tolerance for the isclose function

Should we use the global error tolerance? https://github.com/UXARRAY/uxarray/blob/ff9ec76e5ce3422a8af7005e843fcec0a75155d5/uxarray/constants.py#L13

Reply:
> > I feel like we should set the error tolerance for the isclose function
> 
> Should we use the global error tolerance?
> 
> https://github.com/UXARRAY/uxarray/blob/ff9ec76e5ce3422a8af7005e843fcec0a75155d5/uxarray/constants.py#L13

+1

Reply:
@hongyuchen1030 @rajeeja 

Added the error tolerance and cached flag indicated if we've already run the computation.